tests/Unit/Models/PostModelTest.php

```diff
@@ -153,6 +153,23 @@ public function testBlinkPayload()
         );
     }
 
+    /**
+     * Test expected payload for Customer.io, even if the
+     * post's signup has been deleted.
+     *
+     * @return void
+     */
+    public function testCustomerIoPayloadWithoutSignup()
+    {
+        $post = factory(Post::class)->create();
+        $post->signup->delete();
+
+        $payload = $post->fresh()->toCustomerIoPayload();
+
+        $this->assertArrayHasKey('id', $payload);
+        $this->assertEquals('', $payload['campaign_cause']);
```

Comment:
I should've done this from the get-go 🦎 but this should give us some peace of mind that this doesn't regress in the future (since it's so easy to forget this edge case). The main value of this test is to see whether `toCustomerIoPayload` throws an exception, so I haven't filled out too many assertions besides that.